backend/local/backend_plan.go

```diff
@@ -114,7 +114,12 @@ func (b *Local) opPlan(
 
 	// Save the plan to disk
 	if path := op.PlanOutPath; path != "" {
-		plan.Backend = *op.PlanOutBackend
+		if op.PlanOutBackend != nil {
```

Comment:
I am fairly positive this isn't the proper solution, but this statement was causing a panic.

Reply:
This seems like a reasonable fix to start, to fix the panic. As you suspected, I think it'd be good also to track down why this was `nil` in some cases... maybe it's okay for it to be nil sometimes, but when I wrote this I expected that there'd always be _something_ here, even if it was a synthetic configuration object containing the default settings for the local backend. Once we get things in a more workable state it'll be easier to test what the behavior is with various backends here and see if we need to make any further adjustments.

Reply:
As far as I could tell from the previous version of this code (aka master), the difference is assigning a nil pointer to a value instead of a nil value to a pointer, if that makes sense - I don't think I'm communicating this properly, but there was a slight structural shift to the models. 

Originally `plan.Backend` was a pointer and we'd assign a nil op.PlanOutBackend to that.